rotkehlchen/api/rest.py

```diff
@@ -5388,16 +5388,24 @@ def get_historical_asset_balance(self, asset: Asset, timestamp: Timestamp) -> di
 
     def get_historical_asset_amounts(
             self,
-            asset: Asset,
+            asset: Asset | None,
+            collection_id: int | None,
             from_timestamp: Timestamp,
             to_timestamp: Timestamp,
     ) -> Response:
         try:
-            balances, last_event_identifier = HistoricalBalancesManager(self.rotkehlchen.data.db).get_asset_amounts(  # noqa: E501
-                asset=asset,
-                from_ts=from_timestamp,
-                to_ts=to_timestamp,
-            )
+            if asset is not None:
+                balances, last_event_identifier = HistoricalBalancesManager(self.rotkehlchen.data.db).get_asset_amounts(  # noqa: E501
+                    asset=asset,
+                    from_ts=from_timestamp,
+                    to_ts=to_timestamp,
+                )
+            else:
+                balances, last_event_identifier = HistoricalBalancesManager(self.rotkehlchen.data.db).get_collection_amounts(  # noqa: E501
```

Comment:
can you do `elif collection_id is not None` and have an else that is a bad input? It shouldn't happen since you validate it in the schema but just to be protected